applications/CompressiblePotentialFlowApplication/custom_elements/incompressible_potential_flow_element.cpp

```diff
@@ -572,23 +572,26 @@ void IncompressiblePotentialFlowElement<Dim, NumNodes>::ComputePotentialJump(con
 {
     const array_1d<double, 3> free_stream_velocity = rCurrentProcessInfo[FREE_STREAM_VELOCITY];
     const double free_stream_velocity_norm = sqrt(inner_prod(free_stream_velocity, free_stream_velocity));
+    const double reference_chord = rCurrentProcessInfo[REFERENCE_CHORD];
 
     array_1d<double, NumNodes> distances;
     GetWakeDistances(distances);
 
-    for (unsigned int i = 0; i < NumNodes; i++)
-    {
-        double aux_potential = GetGeometry()[i].FastGetSolutionStepValue(AUXILIARY_VELOCITY_POTENTIAL);
-        double potential = GetGeometry()[i].FastGetSolutionStepValue(VELOCITY_POTENTIAL);
+    auto r_geometry = GetGeometry();
+    for (unsigned int i = 0; i < NumNodes; i++){
+        double aux_potential = r_geometry[i].FastGetSolutionStepValue(AUXILIARY_VELOCITY_POTENTIAL);
+        double potential = r_geometry[i].FastGetSolutionStepValue(VELOCITY_POTENTIAL);
         double potential_jump = aux_potential - potential;
 
-        if (distances[i] > 0)
-        {
-            GetGeometry()[i].SetValue(POTENTIAL_JUMP, -2.0 / free_stream_velocity_norm * (potential_jump));
+        if (distances[i] > 0){
+            r_geometry[i].SetLock();
+            r_geometry[i].SetValue(POTENTIAL_JUMP, - (2.0 * potential_jump) / (free_stream_velocity_norm * reference_chord));
+            r_geometry[i].UnSetLock();
         }
-        else
-        {
-            GetGeometry()[i].SetValue(POTENTIAL_JUMP, 2.0 / free_stream_velocity_norm * (potential_jump));
+        else{
+            r_geometry[i].SetLock();
+            r_geometry[i].SetValue(POTENTIAL_JUMP, (2.0 * potential_jump) / (free_stream_velocity_norm * reference_chord));
```

Comment:
even though safe to access now, isn't this still a race condition?

Reply:
What do you mean by "race" condition?

Reply:
We already discussed it but I write here for the rest. Yes it is, but since the potential jump depends only on the nodal values of the velocity_potential and auxiliary_velocity_potential there is no problem with this regard.

Reply:
> What do you mean by "race" condition?

@philbucher just explained me. A race condition is when the value of a variable depends on which element is the last one to write onto that variable. @philbucher was right in spotting that if potential_jump would be element dependent, the value that we would get would be "random" because it would be dependent on the last element that writes to the node. I don't know if I explained myself correctly. @philbucher you can correct me if I am wrong.

Reply:
Search Results
Race conditions Openmp
The data usage pattern in an OpenMP* parallel region creates a data race. A data race occurs when two threads access the same memory without proper synchronization. ... There are many ways that race conditions can be created. The simplest way is to write to a shared variable in a parallel region.

Reply:
@marcnunezc https://stackoverflow.com/questions/34510/what-is-a-race-condition

Reply:
Thanks for the explanation! I thought it was a typo at first :sweat_smile: